atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractTableMappingService.java

```diff
@@ -57,9 +57,10 @@ public TableReference getMappedTableName(TableReference tableRef) {
             return tableMap.get().get(tableRef);
         } else {
             updateTableMap();
-            Validate.isTrue(tableMap.get().containsKey(tableRef),
-                    "Unable to resolve full name for table reference " + tableRef);
-            return tableMap.get().get(tableRef);
+            if (tableMap.get().containsKey(tableRef)) {
```

Comment:
These seem unsafe... (the old code too)
We should store a local reference to `tableMap.get()` or do a single `get(tableRef)` and then check if it is `null`.